dss/operations/checkout.py

```diff
@@ -190,3 +194,57 @@ def process_keys(self):
                 self._sleepy_checkout(file_checkout, file_metadata=file_metadata, blob_path=blob_path)
             else:
                 sys.stderr.write(f'Invalid key regex: {_key}')
+
+
+@checkout.action("sync",
+                 arguments={"--replica": dict(choices=[r.name for r in Replica], required=True)})
+class Sync(CheckoutHandler):
+    """Checkout all files meeting cache criteria."""
+    def __init__(self, argv: typing.List[str], args: argparse.Namespace):
+        self.keys = []
+        self.replica = Replica[args.replica]
+        self.handle = Config.get_blobstore_handle(self.replica)
+        self.checkout_bucket = self.replica.checkout_bucket
+
+        self.tombstone_cache: Dict[str, bytes] = {}
+        self.tombstone_cache_max_len = 100000
+
+    def _is_file_tombstoned(self, key: str):
+        if key.endswith(f'.{TOMBSTONE_SUFFIX}'):
+            return True
+
+        assert key.startswith(FILE_PREFIX)
+
+        uuid, version = self._parse_key(key)
+        if len(self.tombstone_cache) >= self.tombstone_cache_max_len:
+            self.tombstone_cache.popitem()
+
+        if uuid not in self.tombstone_cache:
+            try:
+                self.tombstone_cache[uuid] = self.handle.get(self.replica.bucket,
+                                                             key=f'{FILE_PREFIX}/{uuid}.{TOMBSTONE_SUFFIX}')
+            except BlobNotFoundError:
+                self.tombstone_cache[uuid] = None
+        return self.tombstone_cache[uuid]
```

Comment:
What’s the purpose of the tombstone cache? Do we expect to list keys more than once?